File: lib/js/src/Fetch__Iterator.js
'use strict';


var Next = { };

function forEach(f, t) {
  while(true) {
    var item = t.next();
    var match = item.done;
    var match$1 = item.value;
    if (match !== undefined && match) {
      if (!(match$1 == null)) {
        return f(match$1);
      } else {
        return ;
      }
    }
    if (match$1 == null) {
      continue ;
    }
    f(match$1);
    continue ;
  };
}

exports.Next = Next;
exports.forEach = forEach;
/* No side effect */

File: src/Fetch__Iterator.ml
module Next = struct
  type 'a t

  external done_ : _ t -> bool option = "done" [@@bs.get]
  external value : 'a t -> 'a option = "value" [@@bs.get] [@@bs.return nullable]
end

type 'a t

external next : 'a t -> 'a Next.t = "next" [@@bs.send]

let rec forEach ~f t =
  let item = next t in
  match Next.(done_ item, value item) with
  | Some true, Some value -> f value [@bs]
  | Some true, None -> ()
  | (Some false | None), Some value ->
    f value [@bs];
    forEach ~f t
  | (Some false | None), None -> forEach ~f t
